onchain/src/art_peace.cairo

```diff
@@ -971,7 +1006,20 @@ pub mod ArtPeace {
             // TODO: Allow some threshold?
             if matches == template_metadata.width * template_metadata.height {
                 self.templates.completed_templates.write(template_id, true);
-            // TODO: Distribute rewards
+                // Distribute rewards
+                if pixel_contributors.len() > 0 {
+                    let mut i = 0;
+
+                    while i < pixel_contributors
+                        .len() {
+                            let mut user = *pixel_contributors.at(i).into();
+                            let user_to_felt: felt252 = user.into();
+                            let total_pixels = total_pixels_by_user.get(user_to_felt);
+
+                            IERC20Dispatcher { contract_address: token }
+                                .transfer(user, total_pixels.into());
```

Comment:
You'll want to transfer a portion of the `reward` from the metadata proportional to the amount of pixels the user placed. So something like `reward * users_total_pixels / total_pixels`. You might also need to do some checks when there is a remainder of reward on the division.